stdlib/3/asyncio/subprocess.pyi

```diff
@@ -9,9 +10,9 @@ if sys.version_info >= (3, 8):
 else:
     _ExecArg = Union[str, bytes]  # Union used instead of AnyStr due to mypy issue  #1236
 
-PIPE: int
-STDOUT: int
-DEVNULL: int
+_S = TypeVar("_S")
+_T = TypeVar("_T")
+_U = TypeVar("_U")
```

Comment:
I think we should constrain these to custom protocols. For example, `stdin` needs to support at least `write()`, `drain()`, and `close()` (possible more).

Reply:
I was basing this on the subprocess code:
https://github.com/python/typeshed/blob/9707ea9bf97e25ae40f596a5973e7c6392ee95af/stdlib/3/subprocess.pyi#L34

That code should probably be constrained to `str`/`bytes`, but that hasn't been done. I figured it was probably not worth doing when this is likely only going to be used internally to this file.

If you think it's worth adding those constraints, I can give it go though.

Reply:
I think if you wanted to have these constraints useful to someone (on the rare occasion) instantiating `Process` directly, it would also become more complex to add the type annotations, as the type are not passed directly, but are part of `protocol`. Therefore we would also need to be `protocols.BaseProtocol` generic as well in order to type check it.

i.e. something like this:
```
class(Generic[_S, _T, _U]):
    def __init__(self, transport, protocol: protocols.BaseProtocol[_S, _T, _U]):
```

Reply:
To make things more complicated, `protocols.BaseProtocol` doesn't contain the `stdin`/`stdout`/`stderr` attributes which are used in this code. I'm not sure at what point in the code they are added, but it suggests that an arbitrary `BaseProtocol` may not even be valid on this class.

Reply:
@srittau Does this make sense to just leave it like this for now?

Reply:
Yes, it's fine as it is.